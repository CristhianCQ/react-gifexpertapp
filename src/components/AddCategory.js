import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [InputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        //  console.log(e.target.value);
          
         setInputValue(e.target.value);
        //  console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
         e.preventDefault();  

         if (InputValue.trim().length > 2) {
            setCategories(cats=> [ InputValue,...cats]);
             // props.setCategories(cats=> [...cats, InputValue]);
            setInputValue('') ;
         }

        
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ InputValue}</p>
            <input
                type="text"
                value={InputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired  
}