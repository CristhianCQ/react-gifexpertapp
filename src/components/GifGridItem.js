import React from 'react'
import PropTypes from 'prop-types'; //impt

export const GifGridItem = ({id, title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}



/* 
    1. enzyme https://enzymejs.github.io/enzyme/ npm i --save-dev enzyme enzyme-adapter-react-16
    2. Enzyme to Jdon https://www.npmjs.com/package/enzyme-to-json npm install --save-dev enzyme-to-json
    3. debe mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSnap
*/