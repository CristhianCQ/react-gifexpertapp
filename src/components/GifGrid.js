 import React from 'react'
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
 import { GifGridItem } from './GifGridItem';
import PropsTypes from 'prop-types'; //impt

export const GifGrid = ({category}) => {


    const {data:images, loading} = useFetchGifs(category);
 
    // const [images, setImages] = useState([]);


    // useEffect( () => {
    //     getGifs(category).then(setImages);  
    // },[category]);
 
     

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {/* {loading? <p> Loading ...</p>: null    } */}
            {loading && <p className="animate__animated animate__flash"> Loading ...</p> }
            <div className="card-grid">
                
                {
                    images.map((img) => (
                        <GifGridItem 
                            key={img.id}
                            {... img}
                        />
                    ))
                }
                
            </div>
        </>
    )
}

GifGrid.propsTypes = {
    category: PropsTypes.string.isRequired
}