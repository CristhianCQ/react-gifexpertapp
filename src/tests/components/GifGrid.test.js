import React from 'react';
import '@testing-library/jest-dom'; 
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs' )

describe('Pruebas en <GifGrid />', () => {
    
    const category = "One punch";


    test('debe de obtener el snapshot del componente montado', () => {
        //I need to mock the data - necesito falcear la data 
        useFetchGifs.mockReturnValue({
            //se debe poner el valor por defecto o inical de hook
            data: [],
            loading: true,
        })


        const wrapper= shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imagenes con nuestro custom hook ', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:',
            title: 'Cualquier Cosa'
        },
        {
            id: '123',
            url: 'https://localhost:',
            title: 'Cualquier Cosa'
        } ]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        })         

        const wrapper= shallow(<GifGrid category={category}/>);

         expect(wrapper).toMatchSnapshot();
         expect(wrapper.find('p').exists()).toBe(false);
         expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
        
})
