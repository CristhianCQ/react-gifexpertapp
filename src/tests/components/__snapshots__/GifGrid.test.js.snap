// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes con nuestro custom hook  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost:"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier Cosa"
      url="https://localhost:"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de obtener el snapshot del componente montado 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
